neuralpredictors/layers/readouts/gaussian.py

```diff
@@ -380,6 +389,23 @@ def feature_l1(self, reduction="sum", average=None):
         else:
             return 0
 
+    def adaptive_l1_lognorm(self, reduction="sum", average=None):
+        if self._original_features:
+            features = self.adaptive.abs() * self.features
+            return self.apply_reduction(features.abs(), reduction=reduction, average=average)
+        else:
+            return 0
+
+    def regularizer(self, reduction="sum", average=None):
+        if self.regularizer_type == "l1":
+            return self.feature_l1(reduction=reduction, average=average) * self.feature_reg_weight
+        else:
+            new_gamma = self.feature_reg_weight
+            readout_reg = self.adaptive_l1_lognorm(reduction=reduction, average=average) * new_gamma
+            #             gammas are supposted to be from lognorm distribution
+            gamma_prior = 1 / (self.gamma_sigma**2) * ((torch.log(self.adaptive.abs()) ** 2).sum())
+            return readout_reg + gamma_prior
+
```

Comment:
see the comment above:)

Reply:
if you are worried that it could be changed after initialisation, I can change the naming `self.regularizer_type` ->`self._regularizer_type` showing that its a private variable

Reply:
I like the idea to use protected variables, but I'd rather not do it here to be consistent with the code we already have, wdyt?

The concern you mentioned is one concern I have. The other concern is that if a new developer starts to add functionality, but messes smth up, it would be nice if the code would throw an error so it is easy to debug, instead of just doing something else per default. This rationale is for example also described here: https://goodresearch.dev/docs.html?highlight=raise#raise-errors

Reply:
which consistency do you mean? because we do have protected variables naming in neuropredictors ([example](https://github.com/sinzlab/neuralpredictors/blob/main/neuralpredictors/layers/readouts/gaussian.py#L301-L304) from readouts) and the `self.regularizer_types` is introduced in this PR, so there would be no back compatibility problems

Reply:
Yes, just most attributes except for a few are marked public and not protected/private. I am also happy to mark them protected here, though this does not address both of my concerns while the NotInolementedError would

Reply:
Yes, I think if a variable if protected (as it is from the yesterday update) then its enough to do the checks during initialisation (as you left the choice for me, at least thats how I read the previous message above wrt throwing errors)

Reply:
That comment you just referred to was regarding another detail. Here I would highly encourage to add the final else. It makes the code safer, easier to debug, easier to read, with just two lines of code. What speaks against it?

Reply:
I do not agree but I do not see a big point in arguing, so have added the error